src/gui/DatabaseOpenWidget.ui

```diff
@@ -31,27 +31,308 @@
+                          <string>Key File:</string>
+                         </property>
+                        </widget>
+                       </item>
+                      </layout>
+                     </item>
+                     <item row="1" column="0">
+                      <layout class="QGridLayout" name="gridLayout">
+                       <property name="verticalSpacing">
+                        <number>0</number>
+                       </property>
+                       <item row="0" column="1">
+                        <widget class="QComboBox" name="comboChallengeResponse">
+                         <property name="enabled">
+                          <bool>false</bool>
+                         </property>
+                         <property name="sizePolicy">
+                          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
+                           <horstretch>0</horstretch>
+                           <verstretch>0</verstretch>
+                          </sizepolicy>
+                         </property>
+                         <property name="editable">
+                          <bool>false</bool>
+                         </property>
+                        </widget>
+                       </item>
+                       <item row="1" column="1">
+                        <widget class="QProgressBar" name="yubikeyProgress">
+                         <property name="maximumSize">
+                          <size>
+                           <width>16777215</width>
+                           <height>2</height>
+                          </size>
+                         </property>
+                         <property name="minimum">
+                          <number>0</number>
+                         </property>
+                         <property name="maximum">
+                          <number>0</number>
+                         </property>
+                         <property name="value">
+                          <number>-1</number>
+                         </property>
+                         <property name="textVisible">
+                          <bool>false</bool>
+                         </property>
+                        </widget>
+                       </item>
+                       <item row="0" column="2">
+                        <widget class="QPushButton" name="buttonRedetectYubikey">
+                         <property name="enabled">
+                          <bool>true</bool>
+                         </property>
+                         <property name="text">
+                          <string>Refresh</string>
+                         </property>
+                        </widget>
+                       </item>
+                       <item row="0" column="0">
+                        <widget class="QLabel" name="label_4">
+                         <property name="text">
+                          <string>YubiKey:</string>
```

Comment:
Recommend naming this "Hardware Token" and add a help tooltip that shows the following when hovered:

> KeePassXC supports Yubikey and OnlyKey hardware tokens with slots configured with HMAC challenge-response. To read more about this solution, please visit https://keepassxc.org/docs#hwtoken.

OnlyKey support is being added to 2.5.0 (eventually)

Reply:
Agree, although "Hardware Token" is quite arcane. Any easier and less technical terms? We had "Challenge Response" before, but that's even harder to understand.

Reply:
"Security Key"
"Multi-Factor Token"

Reply:
I think both "key" and "token" are quite ambiguous. "key" probably even more than "token", which is more of a foreign word to lay users I guess.

Reply:
But if you own one you know the term (which is the point) and the help tooltip helps educate the lay user.

Something like this:

![img](https://www.proprofs.com/knowledgebase/examples/tour_context2.png?v=1)